medicines/api/src/schema.rs

```diff
@@ -2,25 +2,25 @@ use juniper::{FieldResult, RootNode};
 
 use crate::{
     azure_search::AzureContext,
-    product::{get_products_by_substance_name, Product},
-    substance::{get_substances, Substances},
+    product::get_substance_with_products,
+    substance::{get_substances, Substance, Substances},
 };
 
 pub struct QueryRoot;
 
 #[juniper::graphql_object(Context = AzureContext)]
 impl QueryRoot {
-    async fn products(
+    async fn substance(
         context: &AzureContext,
-        substance_name: Option<String>,
-    ) -> FieldResult<Vec<Product>> {
-        match substance_name {
+        name: Option<String>,
```

Comment:
is it possible for this to not be optional and therefore not have a need to handle the None arm or is this because you  could make the call from the client this way so you want to be explicit?

Reply:
if this isn't `Option`, then it becomes a required field in the GraphQL